cmd/erasure-sets.go

```diff
@@ -267,13 +270,22 @@ func (s *erasureSets) monitorAndConnectEndpoints(ctx context.Context, monitorInt
 	// Pre-emptively connect the disks if possible.
 	s.connectDisks()
 
+	monitor := time.NewTimer(monitorInterval)
+	defer monitor.Stop()
+
 	for {
 		select {
 		case <-ctx.Done():
 			return
-		case <-time.After(monitorInterval):
+		case <-monitor.C:
 			s.connectDisks()
 		}
+
+		if !monitor.Stop() {
```

Comment:
Not needed. The only way it can get here is if it fired.

Reply:
It will get here once connectDisks() finishes @klauspost i.e it exhausts all its s.endpoints

Reply:
that is the only safe way to Reset() is by stopping an unfired timer @klauspost